Jenkinsfile

```diff
@@ -0,0 +1,94 @@
+// vim: et:ts=2:sw=2
+// This Jenkinsfile is intended to be used with a companion jenkins-job-builder
+// definition. Aside from the credentials it requires which are accessed in the
+// 'environment' section of this pipeline, it requires the following parameters:
+//   AWS_DOMAIN
+//   AWS_PROFILE
+//   AWS_REGION
+//   CLUSTER_USER
+// It may also provide these optional parameters to override the framework's
+// defaults:
+//   ROOK_IMAGE
+//   CEPH_IMAGE
+pipeline {
+  agent { node { label "rhel-8-large" }}
+  environment {
+    AWS_SHARED_CREDENTIALS_FILE = """${sh(
+      returnStdout: true,
+      script: 'echo $WORKSPACE/.aws/credentials'
+      )}"""
+    AWS_CONFIG_FILE = """${sh(
+      returnStdout: true,
+      script: 'echo $WORKSPACE/.aws/config'
+      )}"""
+    AWS_ACCESS_KEY_ID = credentials('openshift-dev-aws-access-key-id')
+    AWS_SECRET_ACCESS_KEY = credentials('openshift-dev-aws-secret-access-key')
+    PULL_SECRET = credentials('openshift-pull-secret')
```

Comment:
you should use withCredentials binding only on particular places where you really need to have those variables. See examples here:  https://jenkins.io/doc/pipeline/steps/credentials-binding/

I think that you need this pull secret only for deploy so no need to give this PULL_SECRET to all the other parts for example where someone can inject in code?

Reply:
Does `withCredentials` work in declarative pipelines? If so, I agree that it should be used to scope this particular secret to the setup stage.

Reply:
Yes, `withCredentials` works in declarative pipeline.

Reply:
I think it should work